scripts/common.js

```diff
@@ -0,0 +1,31 @@
+const { execSync } = require("child_process");
+
+module.exports = {
+    run: function run(command) {
+        return execSync(command).toString().trim()
+    },
+    getCommitId: function getCommitId() {
+        return this.run('git rev-parse --short HEAD');
+    },
+    determineVersion: function determineVersion() {
+        // Form version string - branch name and commit id
+        let commitId = this.getCommitId();
+        let branchName;
+        if (process.env.BRANCH_NAME) {
+            console.log('Using BRANCH_NAME from the environment')
+            branchName = process.env.BRANCH_NAME
+        } else {
+            console.log('Using local Git repo to determine branch name (environment BRANCH_NAME may also be used)')
+            branchName = this.run('git branch --show-current').replace(/\//g,"-")
```

Comment:
are there any other characters than `/` that are allowed in branch names that would break this?

Reply:
There might be others, but I don't think we're in the habit of using them and `/` is the only character that the Buildkite agents replace when creating the `BRANCH_NAME` environment variable.  So I went with the same approach primarily for consistency.